src/Tribe/Aggregator/Tabs/Abstract.php

```diff
@@ -102,6 +102,36 @@ public function handle_submit() {
 			'schedule_time' => empty( $data['schedule_time'] ) ? null : $data['schedule_time'],
 		);
 
+		// Only apply this verification when dealing with Creating new items
+		if ( ! empty( $post_data['action'] ) && 'new' === $post_data['action'] ) {
+			$hash = array_filter( $meta );
+
+			// remove non-needed data from the Hash of the Record
+			unset( $hash['schedule_day'], $hash['schedule_time'] );
+			ksort( $hash );
+			$hash = maybe_serialize( $hash );
+			$hash = md5( $hash );
+
+			$matches = Tribe__Events__Aggregator__Records::instance()->query( array(
+				'post_status' => Tribe__Events__Aggregator__Records::instance()->get_status( 'schedule' )->name,
+				'meta_query' => array(
+					Tribe__Events__Aggregator__Records::instance()->prefix_meta( 'source' ) => $meta['source'],
```

Comment:
I don't know EA well, but can we just pull the posts here instead of the IDs - since we're going to pull the posts _individually_ immediately after?

Reply:
Unfortunately not really due to how each record needs to be initialized by a different class depending on the source.